site/content/entry/yuzu-progress-report-nov-2023/index.md

```diff
@@ -0,0 +1,359 @@
+
+## Applets, input, and other services
+
+Back this month for more is [Macj0rdan](https://github.com/Macj0rdan) along with german77, partners in crime working in implementing the {{< gh-hovercard "12011" "controller menu applet." >}}
+Functionality is limited at the moment, but you can now see your connected controllers just as intended.
+As expected, a [firmware dump](https://rena21.github.io/yuzu-wiki/setup-guide/dump-firmware/) is required.
+
+{{< single-title-imgs
+    "While this is native fun, don’t expect it to be very usable for the time being"
+    "./applet1.png"
+    "./applet2.png"
+    "./applet3.png"
+    >}}
+
+Working as a solo this time, german77 managed to {{< gh-hovercard "11969" "fix the profile pics" >}} for all currently available system applets, like for example during profile selection.
+The Switch wants exactly 256x256 images, so included as part of the package is a resizer to just give the applet what it wants.
+
+{{< imgs
+	"./profiles.png| They’re multiplying!"
+  >}}
+
+Continuing his [previous work](https://yuzu-emu.org/entry/yuzu-progress-report-jul-2022/#input-improvements) on infrared shenanigans which thankfully don’t involve explosives or self-guidance, german77 finished {{< gh-hovercard "11980" "implementing the" >}} `moment image processor`.
+This means games like `Trombone Champ`and `WarioWare: Put a lid on it` are now playable with Joy-Cons.
+Feel free to crack that volume dial and torture your neighbours to your heart’s content.
+Here’s an example by german77 himself, enjoy:
+
+{{< imgs
+	"./trombone.mp4|Guitar Hero got nothing on this (Trombone Champ)"
+  >}}
+
+Switching a bit to input, german77 fixed an omission in the input code regarding the {{< gh-hovercard "12007" "single Joy-Con SL and SR" >}} buttons, allowing one of the minigames of `WarioWare: Plug your leaks` to be playable. Oops.
+
+{{< imgs
+	"./leak.png|Got teflon tape? (WarioWare: Plug your leaks)"
+  >}}
+
+This breaks backwards compatibility with saved input profiles of single right Joy-Cons―an unlikely configuration for most games. If you are using this configuration, you might have to rebind your controller, but other profiles are unaffected.
+
+Thanks to the work of newcomer [dima-xd](https://github.com/dima-xd), the native {{< gh-hovercard "11929" "software keyboard applet" >}} can be run. 
+Sadly, it’s not possible to use it at this moment as it hasn’t been hooked into the code yet, multiprocess support is needed here.
+As an apology, here’s how it looks:
+
+{{< imgs
+	"./osk.png| Tactile switches? Touch Switch!"
+  >}}
+
+Since yuzu can now deal with Mii data inside Amiibos, the code will also automatically {{< gh-hovercard "12183" "validate their integrity" >}}.
+Thanks to german77’s work, now the emulator can detect corrupted Amiibos and load a backup.
+This avoids crashes in games sensitive to Amiibo integrity, like `Super Smash Bros. Ultimate`.
```

Comment:
```suggestion
Since yuzu can now work with Mii data inside Amiibos, the code will also automatically {{< gh-hovercard "12183" "validate their integrity" >}}.
Thanks to german77’s work, the emulator can now detect corrupted Amiibos and automatically load a backup.
This avoids crashes in games sensitive to Amiibo integrity, like `Super Smash Bros. Ultimate`.
```